pkg/operation/shoot/shoot.go

```diff
@@ -237,9 +237,12 @@ func (s *Shoot) ComputeAPIServerURL(runsInSeed, useInternalClusterDomain bool) s
 	if runsInSeed {
 		return common.KubeAPIServerDeploymentName
 	}
-
+	apiServerPortSuffix := ""
+	if s.CloudProvider == gardenv1beta1.CloudProviderLocal {
+		apiServerPortSuffix = ":31443"
+	}
 	if dnsProvider := s.Info.Spec.DNS.Provider; dnsProvider != nil && *dnsProvider == gardenv1beta1.DNSUnmanaged {
```

Comment:
I would expect that instead of having to check for the DNS provider, `useInternaClusterDomain` should be set to true when DNS provider is `DNSUnmanaged`

Reply:
I think the intention behind this is to have a "stable" fallback because the internal domain is in Gardener's responsibility.

Reply:
Yes, but as there's already the `useInternalClusterDomain` we could reuse it instead of introducing additional logic.